### DISCUSSION TITLE
Cannot build from released tarball

### DISCUSSION THREAD

**Discussion by mdorier** [Tencent/tquic#365] [General]
I downloaded the tarball for version 1.0.0 at https://github.com/Tencent/tquic/releases, then followed the instructions from https://tquic.net/docs/getting_started/installation (replacing git clone with untaring the tarball). The build fails with the following error:

```
CMake Error: The source directory "/home/ubuntu/code/tquic-1.0.0/deps/boringssl" does not appear to contain CMakeLists.txt.
```

It looks like the specified directory (boringssl) is empty in the archive. It's a submodule in the repo. Github doesn't include submodules when releasing archives.

Can you either provide release archives that include boringssl, or update the documentation with instructions on how to get it and build release archives?

**Comment by iyangsj** [comment]
Because of GitHub's limitations, the released tarball can't currently include git submodule code. We suggest using the method below to get the specified release version：

```
# Please replace the 'vx.y.z' with the version you need
git clone -b vx.y.z --recursive https://github.com/Tencent/tquic.git
```

This issue is already mentioned in the [documentation](https://tquic.net/docs/faq/installation/#failed-to-build-the-sources-released-on-github). Thanks for your feedback.
